Project assembled

okonoshi opened the pull request:
Вышло довольно долго и напряжно. Некоторые компоненты работали не так как должны работать в оригинале, некоторые компоненты пришлось переписывать. Реализуешь что-то одно, ломается что то другое, путанится в экспортах и импортах.
Но вроде бы что то вышло

okonoshi committed:
completed dashboard page

okonoshi committed:
add all components. Completed dashboard page

okonoshi committed:
deleted public folders

okonoshi committed:
sidebar  toggler on document

okonoshi committed:
remove console.log

okonoshi committed:
notiffication

okonoshi committed:
Обновил вывод оповещения.
По какой то причине не загружается изображение с ошибкой 403.
А так же при загрузке страницы 2 раза запрос отправляется на сервер.

okonoshi committed:
Проект собран, не все функции реализованы
Работает выбор диапазона дат.
Добавилена возможность скрывать панель навигации

загрузка данных : В оригинальном проекте, при выбор даты,подгружаются
данные только в выбранном диапазоне. И при скроле уже нет подгрузки.
Т.к. в нашем компоненте sortableTable
не учитывается возможность указания даты, в моем варианте подгрузка происходит
без учета диапазона дат. То есть, при скроле загружаются все данные.
Я не смог придумать как изменить sortableTable для работы с датами.Т.к. в одном месте
используются параметры from, to, а в другом createedAT_gte, createdAt_lte.
И не знаю как отключить загрузки при прокрутке, когда указан диапазон.

Раздел продажи : я его тоже делал через sortableTable, но sortableTable
делает ряды ссылками, а для раздела продажи нужно делать в div.
По этомк в продажах клики по заказам переходят на страницу продуктов.
Нужно еще один sortableTable делать для этого?

okonoshi committed:
Поменял ссылку

okonoshi committed:
Реализовал сортировку по названию, дабл слайдер и статус заказа.
Вроде бы справился с подгрузкой товара после указания диапазона, но
не уверен что правильно. Я просто снимаю с документа слушатель скрола.
Мне показалось что именно так работает в оригинальном проекте.
Правда полоса загрузки все равно есть.

okonoshi commented:
Забыл добавить.
Фотографии так и не загружаются. Ошибка 429.  Cannot read properties of undefined (reading 'replace').
Мы пытаемся создать блок с картинкой, а там в escapeHtml попадаетundefined.
Если убрать escapeHtml , то вроде бы 403 ошибка
И при загрузке страницы, если вывести что то в консоль, то два раза срабатывает, как будто страница 2 раза загружается

dosandk commented on src/components/categories/index.js:
```diff
@@ -0,0 +1,173 @@
+export default class SortCategories {
```
внутри данной компоненты посвященной категориям следует использовать компонент "SortableList" к-й мы использовали для drag-and-drop картинок в форме

okonoshi replied:
Я изначально так и думал, но меня запутало то что в компонентах есть отдельная папка categories. Я не понял, нужно ее использовать или нет, по этому решил отдельно продублировать

dosandk commented on src/components/close-panel/index.js:
```diff
@@ -0,0 +1,13 @@
+export default class ClosePanel {
+  initialize() {
+    this.toggleButton = document.querySelector('.sidebar__toggler');
```
🔴 please don't do this
так делать нельзя! компонент должен создавать эл-т, а не получать ссылку на него с помощью поиска

dosandk commented on src/components/column-chart/index.js:
```diff
@@ -1,89 +1,130 @@
+import fetchJson from '../../utils/fetch-json.js';
+
+const BACKEND_URL = 'https://course-js.javascript.ru';
```
Пожалуйста воспользуйтесь переменными окружения: `process.env...`

dosandk commented on src/components/column-chart/index.js:
```diff
@@ -1,89 +1,130 @@
-        </div>
-      </div>
-    `;
+    try {
```
good 👍

dosandk commented on src/components/double-slider/index.js:
```diff
@@ -0,0 +1,194 @@
+export default class DoubleSlider {
```
good 👍

dosandk commented on src/components/notification/index.js:
```diff
@@ -0,0 +1,58 @@
+    `;
+  }
+
+  show(targetElement = document.body) {
```
good 👍

dosandk commented on src/components/notification/index.js:
```diff
@@ -0,0 +1,58 @@
+  destroy() {
+    this.remove();
+    this.element = null;
+    if (NotificationMessage.previousCall === this) {
```
Эта проверка лишняя

okonoshi replied:
В моем решении, без этой проверки, при повторной генерации оповещения, они начинают дублироваться. Точнее старые не удаляются.
При каждом вызове мы перезаписываем NotificationMessage.previousCall = this;
Если удалить проверку, то при вызове show не будет работать это условие
if (NotificationMessage.previousCall) {
      NotificationMessage.previousCall.element.remove();
    }

dosandk commented on src/components/product-form/index.js:
```diff
@@ -1,278 +1,154 @@
 import fetchJson from '../../utils/fetch-json.js';
+import iconGrab from '../../assets/icons/icon-grab.svg';
+import iconTrash from '../../assets/icons/icon-trash.svg';
+const IMGUR_CLIENT_ID = '28aaa2e823b03b1';
```
используйте переменные окружения!

dosandk commented on src/components/product-form/index.js:
```diff
@@ -1,278 +1,154 @@
+    const input = document.createElement('input');
+    input.type = 'file';
+    input.accept = 'image/*';
+    input.addEventListener('change', async () => {
```
👍 good

dosandk commented on src/components/product-form/index.js:
```diff
@@ -1,278 +1,154 @@
+    input.click();
+  };
+
+  sendFormBtnClick = async event => {
```
👍 good 
можно просто "sendForm"

dosandk commented on src/components/product-form/index.js:
```diff
@@ -1,278 +1,154 @@
-    `;
-  }
-
   async render() {
```
good 👍

dosandk commented on src/components/product-form/index.js:
```diff
@@ -284,64 +160,155 @@ export default class ProductForm {
 
-    for (const item of elements) {
-      subElements[item.dataset.element] = item;
+  async uploader(file) {
```
good 👍

dosandk commented on src/components/product-form/index.js:
```diff
@@ -284,64 +160,155 @@ export default class ProductForm {
-    const sortableList = new SortableList({
-      items
-    });
+    const sortableList = new SortableList({ items });
```
переиспользуйте данный компонент SortableList для категорий

dosandk commented on src/components/sortable-table/index.js:
```diff
@@ -1,4 +1,4 @@
-import fetchJson from "../../utils/fetch-json.js";
+import fetchJson from '../../utils/fetch-json.js';
 
 const BACKEND_URL = 'https://course-js.javascript.ru';
```
Переменные окружения!

dosandk commented on src/components/tooltip/index.js:
```diff
@@ -1,78 +1,65 @@
 
-  onMouseMove = event => {
-    this.moveTooltip(event);
+    document.addEventListener("pointermove", this.handlePointerMove);
```
👍 good

dosandk commented on src/components/tooltip/index.js:
```diff
@@ -1,78 +1,65 @@
-    document.removeEventListener('pointerout', this.onMouseOut);
-    this.removeTooltip();
+    this.remove();
+    document.removeEventListener("pointerover", this.handlePointerOver);
```
👍 good

dosandk commented on src/index.html:
```diff
@@ -1,55 +1,55 @@
+      </ul>
+    </aside>
+    <section class="content" id="content"></section>
+    <script type="module" src="/index.js"></script>
```
почему тут подключен `index.js`?
это должно происходить автоматически с помощью webpack

boris-catsvill commented on src/components/column-chart/index.js:
```diff
@@ -1,89 +1,130 @@
-  async render() {
-    const element = document.createElement('div');
+  async update(from, to) {
+    const fromDate = from.toISOString().split('T')[0];
```
```js
const [fromDate] = from.toISOString().split('T');
```

boris-catsvill commented on src/components/notification/index.js:
```diff
@@ -0,0 +1,58 @@
+export default class NotificationMessage {
+  static previousCall;
```
лучше дать имя `previousInstance`

boris-catsvill commented on src/components/product-form/index.js:
```diff
@@ -1,278 +1,154 @@
+
+  sendFormBtnClick = async event => {
+    event.preventDefault();
+    const url = new URL('/api/rest/products', BACKEND_URL);
```
`url` можно сформировать заранее, например в конструкторе, чтобы не делать это при каждой отправке формы

boris-catsvill commented on src/components/tooltip/index.js:
```diff
@@ -1,78 +1,65 @@
+    this.remove();
+  };
 
   constructor() {
```
good 👍

boris-catsvill commented on src/index.js:
```diff
@@ -1,8 +1,9 @@
+import ClosePanel from './components/close-panel';
 
+new Tooltip().initialize();
+new ClosePanel().initialize();
```
может стоит создать тут компонент? например App
и поместить в него шоблон из `index.html` и уже внутри инициализировать тултип и панель

boris-catsvill commented on src/pages/categories/index.js:
```diff
@@ -0,0 +1,152 @@
+import SortCategories from '../../components/categories/index.js';
+import fetchJson from '../../utils/fetch-json.js';
+import NotificationMessage from '../../components/notification/index.js';
+const BACKEND_URL = 'https://course-js.javascript.ru/';
```
! переменные окружения

boris-catsvill commented on src/pages/categories/index.js:
```diff
@@ -0,0 +1,152 @@
+  }
+}
+{
+  /* <div class="notification notification_success show" style="--value:
```
почему этот код закомментирована?

boris-catsvill commented on src/pages/dashboard/index.js:
```diff
@@ -5,147 +5,148 @@ import header from './bestsellers-header.js';
 
 import fetchJson from '../../utils/fetch-json.js';
 
+const BACKEND_URL = 'https://course-js.javascript.ru/';
```
! переменные окружения

boris-catsvill commented on src/pages/dashboard/index.js:
```diff
@@ -5,147 +5,148 @@ import header from './bestsellers-header.js';
-
-      root.append(element);
-    });
+  loadData(from, to) {
```
good 👍

boris-catsvill commented on src/pages/products/edit/index.js:
```diff
@@ -1,18 +1,74 @@
+    });
 
-    return this.element;
+    this.form.addEventListener('error', () => {
```
👍 good

boris-catsvill commented on src/pages/sales/index.js:
```diff
@@ -0,0 +1,106 @@
+      isSortLocally: false
+    });
+
+    this.components = {
```
good 👍

boris-catsvill commented on src/pages/sales/index.js:
```diff
@@ -0,0 +1,106 @@
+    this.components.sortableTable.update(response);
+  }
+
+  loadData(from, to) {
```
good 👍

boris-catsvill reviewed (approved):
good 👍

okonoshi committed:
использовал sortableList в категориях

okonoshi committed:
отправка отсортированных данных на сервер

okonoshi committed:
небольшие правки. Создал компонент App который
генерирует главную страницу.
Компонент close-panel переименовал теперь создает кнопку закрытия

okonoshi committed:
deleted render

okonoshi commented:
Внес изменения. Изображения к сожалению так и не загружаются. Не знаю в чем проблема(

dosandk commented on src/components/app/index.js:
```diff
@@ -0,0 +1,47 @@
+import ClosePanel from '../close-panel/index.js';
+import header from './sidebar-config.js';
+
+export default class App {
```
good 👍

dosandk commented on .env:
```diff
@@ -1,2 +1,2 @@
-BACKEND_URL = https://course-js.javascript.ru/
```
good 👍

dosandk commented on src/components/app/sidebar-config.js:
```diff
@@ -0,0 +1,27 @@
+const header = [
```
good 👍

dosandk commented on src/components/product-form/index.js:
```diff
@@ -3,8 +3,8 @@ import escapeHtml from '../../utils/escape-html.js';
 import iconTrash from '../../assets/icons/icon-trash.svg';
-const IMGUR_CLIENT_ID = '28aaa2e823b03b1';
-const BACKEND_URL = 'https://course-js.javascript.ru';
+const IMGUR_CLIENT_ID = process.env.IMGUR_CLIENT_ID;
```
good 👍
